connect/runtime/src/main/java/org/apache/kafka/connect/util/KafkaBasedLog.java

```diff
@@ -67,6 +69,19 @@
  * </p>
  */
 public class KafkaBasedLog<K, V> {
+
+    /**
+     * A functional interface to obtain the description of a topic with a given name, creating the topic if necessary.
+     */
+    public interface TopicSupplier {
+        /**
+         * Get the description of the topic with the given name, optionally creating the topic if it does not yet exist.
+         * @param topicName the name of the topic
+         * @return the description for the newly-created or existing topic; null if it could not be obtained
+         */
+        TopicDescription getOrCreateTopic(String topicName);
+    }
+
```

Comment:
@cmccabe Here's the interface in Kafka Connect that uses the `TopicDescription` class. Code that uses `KafkaBasedLog` will provide their own implementation of `TopicSupplier`, and that way `KafkaBasedLog` can use this when necessary to create topics without having any knowledge of _how_ that is done. The challenge then becomes how to test `KafkaBasedLog`, since the unit test will need to provide `TopicSupplier` implementations that return `TopicDescription` instances; see [here](https://github.com/apache/kafka/pull/2984/files#diff-28145281473ec8c8a817252d6dbb82b9R110). This is where the public constructor on `TopicDescription` (and `TopicPartitionInfo`) come in very handy and simplify things a great deal.